src/Symfony/Component/Cache/Traits/RedisTrait.php

```diff
@@ -368,6 +370,12 @@ protected function doClear($namespace)
         }
 
         foreach ($this->getHosts() as $host) {
+            if ($host instanceof ClientInterface && $host->getConnection() instanceof ReplicationInterface && method_exists($host, 'getClientFor')) {
```

Comment:
this would be cleaner I think:
`if ($host instanceof Client && $host->getConnection() instanceof ReplicationInterface) {`

I'm wondering: when we hit the master, there is no need anymore to loop on the other hosts, isn't it?
can we do this check outside of the foreach then, and replace `$this->getHosts()` with `$hosts = [$master]` when this happens?

Reply:
> I'm wondering: when we hit the master, there is no need anymore to loop on the other hosts, isn't it?

Sounds good to me. Thought about multi-master replication, but they are converged asynchronously anyway.